src/Cache/FileCacheAdapter.php

```diff
@@ -0,0 +1,325 @@
+<?php
+/**
+ * PHP Domain Parser: Public Suffix List based URL parsing.
+ *
+ * @see http://github.com/jeremykendall/php-domain-parser for the canonical source repository
+ *
+ * @copyright Copyright (c) 2017 Jeremy Kendall (http://jeremykendall.net)
+ * @license   http://github.com/jeremykendall/php-domain-parser/blob/master/LICENSE MIT License
+ */
+declare(strict_types=1);
+
+namespace Pdp\Cache;
+
+use DateInterval;
+use FilesystemIterator;
+use Generator;
+use Psr\SimpleCache\CacheInterface;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use Traversable;
+
+/**
+ * A simple file-based PSR-16 cache implementation.
+ *
+ * This class is heavily based on the code found in
+ *
+ * @see https://github.com/kodus/file-cache/blob/master/src/FileCache.php
+ */
+final class FileCacheAdapter implements CacheInterface
+{
+    /**
+     * @var string control characters for keys, reserved by PSR-16
+     */
+    const PSR16_RESERVED = '/\{|\}|\(|\)|\/|\\\\|\@|\:/u';
+
+    /**
+     * @var string
+     */
+    private $cache_path;
+
+    /**
+     * @var int
+     */
+    private $default_ttl = 86400;
+
+    /**
+     * @var int
+     */
+    private $dir_mode = 0775;
+
+    /**
+     * @var int
+     */
+    private $file_mode = 0664;
+
+    /**
+     * @param string $cache_path absolute root path of cache-file folder
+     */
+    public function __construct(string $cache_path = '')
```

Comment:
The only thing the FileCacheAdapter allow to change is the root directory in which the content will be cache just like before. The ttl is fixed to 1 DAY ... this means that the cache is updated at least once a day ... is it too much ?